### PR TITLE
[MAINTENANCE] Preparatory cleanup refactoring of get_compute_domain

### PR DESCRIPTION
**Pull request by anthonyburdi**
Changes proposed in this pull request:
- Refactor duplicated code for splitting compute domain in each ExecutionEngine into parent
- Override needed for SqlAlchemyExecutionEngine due to column quoting

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

File: great_expectations/execution_engine/execution_engine.py [hunk 1]
```diff
@@ -517,6 +517,77 @@ class MetricDomainTypes.
...
+        if (
+            domain_type != MetricDomainTypes.TABLE
+            and accessor_keys is not None
+            and len(list(accessor_keys)) > 0
+        ):
+            logger.warning(
+                'Accessor keys ignored since Metric Domain Type is not "table"'
+            )
+
+        compute_domain_kwargs = copy.deepcopy(domain_kwargs)
```
Comment (donaldheppner) [inline great_expectations/execution_engine/execution_engine.py]: Consider moving this assignment to an else clause

File: great_expectations/execution_engine/execution_engine.py [hunk 2]
```diff
@@ -517,6 +517,77 @@ class MetricDomainTypes.
...
+        accessor_domain_kwargs = {}
+        if domain_type == MetricDomainTypes.TABLE:
+            (
+                compute_domain_kwargs,
+                accessor_domain_kwargs,
+            ) = self._split_table_metric_domain_kwargs(
```
Comment (donaldheppner) [inline great_expectations/execution_engine/execution_engine.py]: I think positional arguments here are sufficient

File: great_expectations/execution_engine/execution_engine.py [hunk 3]
```diff
@@ -517,6 +517,77 @@ class MetricDomainTypes.
...
+                domain_kwargs=domain_kwargs,
+                domain_type=domain_type,
+                accessor_keys=accessor_keys,
+            )
+
+        elif domain_type == MetricDomainTypes.COLUMN:
+            (
+                compute_domain_kwargs,
+                accessor_domain_kwargs,
+            ) = self._split_column_metric_domain_kwargs(
```
Comment (donaldheppner) [inline great_expectations/execution_engine/execution_engine.py]: with this level of verbosity, I don't think we've saved ourselves any effort by using a Tuple. I'd consider replacing the return type with a simple class that is more descriptive; the class name could be "SplitDomainKwargs" with attributes for "compute" and "accessor"

**Review by donaldheppner — COMMENTED** [review]
some nits, thoughts. nice refactor

**Review by donaldheppner — APPROVED** [review]

File: great_expectations/execution_engine/execution_engine.py [hunk 4]
```diff
@@ -138,6 +139,17 @@ def resolve_data_reference(
...
 
+@dataclass
+class SplitDomainKwargs:
+    """compute_domain_kwargs, accessor_domain_kwargs when split from domain_kwargs
+
+    The union of compute_domain_kwargs, accessor_domain_kwargs is the input domain_kwargs
+    """
+
+    compute: dict
+    accessor: dict
```
Comment (cdkini) [inline great_expectations/execution_engine/execution_engine.py:150]: Do we know anything more about these dicts or can anything go in them?

Comment (anthonyburdi) [reply great_expectations/execution_engine/execution_engine.py:150]: Good call. In the future these will be DomainKwargs but saving that for a future refactor & coordinate with RBP - I want to be careful as this underpins a lot of GE.

**Pull request merged**